sql/connect/common/src/main/scala/org/apache/spark/sql/connect/KeyValueGroupedDataset.scala

```diff
@@ -632,6 +651,53 @@ private class KeyValueGroupedDatasetImpl[K, V, IK, IV](
     }
   }
 
+  override protected[sql] def transformWithStateHelper[U: Encoder, S: Encoder](
+      statefulProcessor: StatefulProcessor[K, V, U],
+      timeMode: TimeMode,
+      outputMode: OutputMode,
+      initialState: Option[sql.KeyValueGroupedDataset[K, S]] = None,
+      eventTimeColumnName: String = ""): Dataset[U] = {
+    val outputEncoder = agnosticEncoderFor[U]
+    val stateEncoder = agnosticEncoderFor[S]
+    val inputEncoders: Seq[AgnosticEncoder[_]] = Seq(kEncoder, stateEncoder, ivEncoder)
+
+    // SparkUserDefinedFunction is creating a udfPacket where the input function are
```

Comment:
@hvanhovell Please confirm whether this is the right direction. I have no context so I'd like to defer to the committers who work with Spark Connect.

Reply:
This is fine. You could also use getUdf for this.